lib/project_types/extension/models/specification_handlers/checkout_ui_extension.rb

```diff
@@ -62,6 +62,14 @@ def localization(context)
           end
         end
 
+        def read_locale_file(filename)
+          content = File.read(filename, mode: "rt", encoding: "bom|utf-8").strip
```

Comment:
Is my interpretation correct that [this approach](https://ruby-doc.org/core-2.7.0/IO.html#method-c-new-label-IO+Encoding) sets the encoding _based_ on the BOM? So, if the file starts with a [UTF-16 BOM](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-16) then UTF-16 will be the encoding of `content`? In other words, I don't quite know what the `|utf-8` part will do.

Reply:
Yeah, I think the encoding would be based on the BOM in the file and if there's nothing or it could not be figure out, it's going to be `utf-8` by default I believe.